### DISCUSSION TITLE
[suggestion]smartphone interaction and info

### DISCUSSION THREAD

**Discussion by vxsw** [zepp-health/zeppos-forum#334] [Developer Feedback]
hi,
as I'd understood, at this time is no way (at least easy way) to get some sort of smartphone info, such as:
- battery level (include onchange event)
- type, state and level of any connectivity cell/wi-fi/nfc etc. and ability to catch events of it's changes
- display/smartphone state (on/off/aod, locked/unlocked) and event of state changed

even m.b. some sort of smartphone sensors, for sample:
- fingerprint event (onunlock - success/fail)
m.b. something else.

I mean. all of this data may be catched by zeppapp, so would be nice to have an ability to transfer it to the watch. it would be very useful for using the watch as a smartphone co-pilot.

also would be nice to make a queue for messages to the watch and send it only if the watch is visible and may receive it, not just "send and forgot" (m.b. with queue length limitation per source).

**Comment by silver-zepp** [comment]
Hey. This is unrealistic, as to collect all this data our app will require all the permissions possible, for it to work. How will we explain to our users why our app needs those permissions?

What you can do however is to run a local server on your smartphone and use either MacroDroid or Tasker to collect that data, then use [ZML](https://www.npmjs.com/package/@zeppos/zml) (Fetch API wrapper) to send that data to/from your watch.

**Comment by vxsw** [comment]
seems I must to clarify and split my answer in two. ok.

at first - no, it's not.
if we talk about permissions as a main limitation, I've check a few apps and battery informers have no any special permissions. connectivity info seems required only "ACCESS_FINE_LOCATION", but zepp already have it. info about lock screen state do not required any permission too. seems fingerprint required additional permission, but it's not critical. I mean, I've talk mostly about principal instead of exact features. leitmotiv is to be a way closer to smaprtphone co-piloting, not just calling and unread messages.
MacroDroid and Tasker... fine, but it's a personal solution. if we talk about common customer, who would delve into it? who would search and install additional third-party app, to get worked some features (for sample battery level of a smartphone on watch face)? it's not a common customer level.

at second. piece of pie.
from the far:
- smart watches are reveals only with smartphones, so the most potential customers of a smartwatch vendors is a smartphone users or a customer would forced to get smartphone to reveal smartwatch abilities.
- how much smart watch vendors do promote their devices as a sports/health tool? seems most of them from cheapest to luxurious.
- how much smart watch vendors do promote their devices as a smartphone co-pilot, a bit more than caller/messages reader? seems no one.
- how much sportsmans you think in all of a smartphone users?
- how much smartphone users not related to any sport activities you think?
- didn't you think, what you focused on a small enough piece of pie with a lot of concurrents and ignore a lot of pies around?

I can't tell you about spotsmans and usability for them, but for non sportsmans, most of a sport/health features more like a funny toy what would be forgotten after few weeks/month, or would be like a just stats/numbers (something shows and it's ok). most non sportsmans customers use a smart watch just like an accessory, m.b. a part of image. "it must be beauty, customizable enough and show some strange numbers to looks cool/sportage". and a lot of people can't see a reason to buy it, because it's mostly just a toy or even useless for them.
if the smart watches would be a way close and useful to/for usual smartphone users (non sportmans), it'll bee more interesting for them. and I think exactly "co-pilot" functions would be a way more useful for them. just they even not imagine what it even may be possible and how it may works.

I do not suggest you to reprofile from sportsmans to non sportsmans, no - just a bit expansion, what may allow you to get a lot of new customers. you already have a excellent and very flexible platform, I don't think what a bit expansion will cost you so big efforts, but it may allow you to get a lot of new non sportage customers if you'll find a correct way to represent them co-pilot abilities and get them interested. it'll may transform to very big and interesting market segment and will costs mostly nothing for you.
just think about it.

**Comment by vxsw** [comment]
yep, forget about messages queue. it's not require any permissions and looks a way more logically then blind sending. for sample, if you'll leave your smartphone for few minutes and got incoming call/message, after returning and restore connection, would be nice to get skipped message or info about missing call. and to avoid spam messages because of a long time apsense, just limits a maximum queue length.
